tests/handles.test.ts

```diff
@@ -0,0 +1,126 @@
+import { ControllerFactory } from './controllers';
+import { DatabaseConnection, PortalState, UserFactory } from './data';
+
+beforeAll(async () => {
+  await DatabaseConnection.connect();
+});
+
+beforeEach(async () => {
+  await DatabaseConnection.clear();
+});
+
+afterAll(async () => {
+  await DatabaseConnection.clear();
+  await DatabaseConnection.close();
+});
+
+describe('Set User Handle', () => {
+  test('Logged In User Can Successfully Set Their Own Handle', async () => {
+    // Setup Required Data
+    const conn = await DatabaseConnection.get();
+    const member = UserFactory.fake();
+
+    // Persist to Database
+    await new PortalState().createUsers(member).write();
+
+    // Get API Controller
+    const userController = ControllerFactory.user(conn);
+    const handle = 'mycustomhandle';
+
+    // Get Updated User Profile
+    const patchedUserResponse = await userController.patchCurrentUser({ user: { handle } }, member);
+
+    expect(patchedUserResponse.error).toBeNull();
+
+    const patchedUser = patchedUserResponse.user;
+
+    expect(patchedUser).toEqual({ ...member.getFullUserProfile(), handle });
+  });
+
+  test('User Cannot Set Handle Using an Invalid Handle Format', async () => {
+    // TODO: I can't get this test to reject with an error because validators aren't run here for some reason
```

Comment:
Right, the decorators only run when the API is hit via. HTTP (my bad, I should probably add this to the testing guidelines). But what we can do is manually invoke the validator that we want to test, as done here: https://github.com/acmucsd/membership-portal/blob/master/tests/feedback.test.ts#L47-L56